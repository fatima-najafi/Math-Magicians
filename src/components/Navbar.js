import React from 'react';
import { Link } from 'react-router-dom';
import './Style.css';

function Navbar() {
  return (
    <div className="container">
      <div>
        <h1>Math Magicians</h1>
      </div>
      <div className="navbar">
        <div className="nav-item">
          <Link to="/">Home</Link>
        </div>
        <div className="nav-item">
          <span className="nav-item">|</span>
          <Link to="/calculator">Calculator</Link>
        </div>
        <div className="nav-item">
          <span className="nav-item">|</span>
          <Link to="/quote">Quote</Link>
        </div>
      </div>
    </div>
  );
}
export default Navbar;
