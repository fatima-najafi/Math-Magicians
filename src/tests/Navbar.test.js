import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

test('renders learn react link', () => {
  render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>,
  );

  const homeLink = screen.getByText(/Home/i);
  const calculatorLink = screen.getByText(/Calculator/i);
  const quoteLink = screen.getByText(/Quote/i);

  expect(homeLink).toBeInTheDocument();
  expect(calculatorLink).toBeInTheDocument();
  expect(quoteLink).toBeInTheDocument();

  expect(homeLink).toMatchSnapshot();
  expect(calculatorLink).toMatchSnapshot();
  expect(quoteLink).toMatchSnapshot();
});
