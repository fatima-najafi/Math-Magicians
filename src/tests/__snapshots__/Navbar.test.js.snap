// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<a
  href="/"
>
  Home
</a>
`;

exports[`renders learn react link 2`] = `
<a
  href="/calculator"
>
  Calculator
</a>
`;

exports[`renders learn react link 3`] = `
<a
  href="/quote"
>
  Quote
</a>
`;
